File: openbis_api/source/java/ch/ethz/sis/openbis/generic/shared/api/v3/dto/entity/sample/SampleUpdate.java
/*
 * Copyright 2013 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.shared.api.v3.dto.entity.sample;

import java.io.Serializable;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.shared.api.v3.dto.entity.AttachmentListUpdateValue;
import ch.ethz.sis.openbis.generic.shared.api.v3.dto.entity.FieldUpdateValue;
import ch.ethz.sis.openbis.generic.shared.api.v3.dto.entity.IdListUpdateValue;
import ch.ethz.sis.openbis.generic.shared.api.v3.dto.entity.ListUpdateValue.ListUpdateAction;
import ch.ethz.sis.openbis.generic.shared.api.v3.dto.id.experiment.IExperimentId;
import ch.ethz.sis.openbis.generic.shared.api.v3.dto.id.sample.ISampleId;
import ch.ethz.sis.openbis.generic.shared.api.v3.dto.id.space.ISpaceId;
import ch.ethz.sis.openbis.generic.shared.api.v3.dto.id.tag.ITagId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("dto.entity.sample.SampleUpdate")
public class SampleUpdate implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private ISampleId sampleId;

    @JsonProperty
    private FieldUpdateValue<IExperimentId> experimentId = new FieldUpdateValue<IExperimentId>();

    @JsonProperty
    private FieldUpdateValue<ISpaceId> spaceId = new FieldUpdateValue<ISpaceId>();

    @JsonProperty
    private IdListUpdateValue<ITagId> tagIds = new IdListUpdateValue<ITagId>();

    @JsonProperty
    private Map<String, String> properties = new HashMap<String, String>();

    @JsonProperty
    private FieldUpdateValue<ISampleId> containerId = new FieldUpdateValue<ISampleId>();

    @JsonProperty
    private IdListUpdateValue<ISampleId> containedIds = new IdListUpdateValue<ISampleId>();

    @JsonProperty
    private IdListUpdateValue<ISampleId> parentIds = new IdListUpdateValue<ISampleId>();

    @JsonProperty
    private IdListUpdateValue<ISampleId> childIds = new IdListUpdateValue<ISampleId>();

    @JsonProperty
    private AttachmentListUpdateValue attachments = new AttachmentListUpdateValue();

    @JsonIgnore
    public ISampleId getSampleId()
    {
        return sampleId;
    }

    @JsonIgnore
    public void setSampleId(ISampleId sampleId)
    {
        this.sampleId = sampleId;
    }

    @JsonIgnore
    public FieldUpdateValue<IExperimentId> getExperimentId()
    {
        return experimentId;
    }

    @JsonIgnore
    public void setExperimentId(IExperimentId experimentId)
    {
        this.experimentId.setValue(experimentId);
    }

    @JsonIgnore
    public FieldUpdateValue<ISpaceId> getSpaceId()
    {
        return spaceId;
    }

    @JsonIgnore
    public void setSpaceId(ISpaceId spaceId)
    {
        this.spaceId.setValue(spaceId);
    }

    @JsonIgnore
    public FieldUpdateValue<ISampleId> getContainerId()
    {
        return containerId;
    }

    @JsonIgnore
    public void setContainerId(ISampleId containerId)
    {
        this.containerId.setValue(containerId);
    }

    @JsonIgnore
    public void setProperty(String key, String value)
    {
        properties.put(key, value);
    }

    @JsonIgnore
    public Map<String, String> getProperties()
    {
        return properties;
    }

    @JsonIgnore
    public IdListUpdateValue<ITagId> getTagIds()
    {
        return tagIds;
    }

    @JsonIgnore
    public void setTagActions(List<ListUpdateAction<ITagId>> actions)
    {
        tagIds.setActions(actions);
    }

    @JsonIgnore
    public IdListUpdateValue<ISampleId> getContainedIds()
    {
        return containedIds;
    }

    @JsonIgnore
    public void setContainedActions(List<ListUpdateAction<ISampleId>> actions)
    {
        containedIds.setActions(actions);
    }

    @JsonIgnore
    public IdListUpdateValue<ISampleId> getParentIds()
    {
        return parentIds;
    }

    @JsonIgnore
    public void setParentActions(List<ListUpdateAction<ISampleId>> actions)
    {
        parentIds.setActions(actions);
    }

    @JsonIgnore
    public IdListUpdateValue<ISampleId> getChildIds()
    {
        return childIds;
    }

    @JsonIgnore
    public void setChildActions(List<ListUpdateAction<ISampleId>> actions)
    {
        childIds.setActions(actions);
    }

    @JsonIgnore
    public AttachmentListUpdateValue getAttachments()
    {
        return attachments;
    }

    @JsonIgnore
    public void setAttachmentsActions(List<ListUpdateAction<Object>> actions)
    {
        attachments.setActions(actions);
    }

}

File: js-test/servers/common/core-plugins/tests/1/as/webapps/openbis-v3-api-test/html/models/dto/entity/sample/SampleUpdate.js
/**
 *  @author pkupczyk
 */
define(["support/stjs"], function (stjs) {
    var SampleUpdate = function() {};
    stjs.extend(SampleUpdate, null, [], function(constructor, prototype) {
        prototype['@type'] = 'SampleUpdate';
        constructor.serialVersionUID = 1;
        prototype.sampleId = null;
        prototype.experimentId = new FieldUpdateValue();
        prototype.spaceId = new FieldUpdateValue();
        prototype.tagIds = new IdListUpdateValue();
        prototype.properties = new HashMap();
        prototype.containerId = new FieldUpdateValue();
        prototype.containedIds = new IdListUpdateValue();
        prototype.parentIds = new IdListUpdateValue();
        prototype.childIds = new IdListUpdateValue();
        prototype.attachments = new AttachmentListUpdateValue();
        prototype.getSampleId = function() {
            return this.sampleId;
        };
        prototype.setSampleId = function(sampleId) {
            this.sampleId = sampleId;
        };
        prototype.getExperimentId = function() {
            return this.experimentId;
        };
        prototype.setExperimentId = function(experimentId) {
            this.experimentId.setValue(experimentId);
        };
        prototype.getSpaceId = function() {
            return this.spaceId;
        };
        prototype.setSpaceId = function(spaceId) {
            this.spaceId.setValue(spaceId);
        };
        prototype.getContainerId = function() {
            return this.containerId;
        };
        prototype.setContainerId = function(containerId) {
            this.containerId.setValue(containerId);
        };
        prototype.setProperty = function(key, value) {
            this.properties.put(key, value);
        };
        prototype.getProperties = function() {
            return this.properties;
        };
        prototype.getTagIds = function() {
            return this.tagIds;
        };
        prototype.setTagActions = function(actions) {
            this.tagIds.setActions(actions);
        };
        prototype.getContainedIds = function() {
            return this.containedIds;
        };
        prototype.setContainedActions = function(actions) {
            this.containedIds.setActions(actions);
        };
        prototype.getParentIds = function() {
            return this.parentIds;
        };
        prototype.setParentActions = function(actions) {
            this.parentIds.setActions(actions);
        };
        prototype.getChildIds = function() {
            return this.childIds;
        };
        prototype.setChildActions = function(actions) {
            this.childIds.setActions(actions);
        };
        prototype.getAttachments = function() {
            return this.attachments;
        };
        prototype.setAttachmentsActions = function(actions) {
            this.attachments.setActions(actions);
        };
    }, {sampleId: "ISampleId", experimentId: {name: "FieldUpdateValue", arguments: ["IExperimentId"]}, spaceId: {name: "FieldUpdateValue", arguments: ["ISpaceId"]}, tagIds: {name: "IdListUpdateValue", arguments: ["ITagId"]}, properties: {name: "Map", arguments: [null, null]}, containerId: {name: "FieldUpdateValue", arguments: ["ISampleId"]}, containedIds: {name: "IdListUpdateValue", arguments: ["ISampleId"]}, parentIds: {name: "IdListUpdateValue", arguments: ["ISampleId"]}, childIds: {name: "IdListUpdateValue", arguments: ["ISampleId"]}, attachments: "AttachmentListUpdateValue"});
    return SampleUpdate;
})